packages/openapi-generator/src/parser/operation.ts

```diff
@@ -72,82 +72,81 @@ export function getRelevantParameters(
   );
 }
 
-function isPlaceholder(pathPart: string): boolean {
-  return /^\{.+\}$/.test(pathPart);
-}
+/**
+ * @internal
+ */
+export function parsePathPattern(
+  pathPattern: string,
+  pathParameters: OpenApiParameter[]
+): string {
+  // Replace the old placeholder with the name of the corresponding path parameter
+  for (const pathParameter of pathParameters) {
+    // Check if path contains the provided path parameter
+    if (!pathPattern.includes(pathParameter.originalName)) {
+      throw new Error(
+        `Could not find placeholder for path parameter '${pathParameter.originalName}'.`
+      );
+    }
+    // Temporarily using TEMP_LEFT_CURLY_BRACKET and TEMP_RIGHT_CURLY_BRACKET to avoid re-substitution
+    // Check test 'parses path template for parameters' for the edge case input
+    pathPattern = pathPattern.replace(
+      new RegExp(`{${pathParameter.originalName}}`, 'g'),
+      `TEMP_LEFT_CURLY_BRACKET${pathParameter.name}TEMP_RIGHT_CURLY_BRACKET`
+    );
+  }
```

Comment:
I am not a fan of `TEMP_LEFT_CURLY_BRACKET` and I don't think that this part was the problem in the old implementation. The parameter names are unique, aren't they?

The old implementation did the following:
1. split string by '/'
2. filter parts that have placeholders
4. check that every part includes at most one `{placeholders}`

(with some other steps between and after)

I think the problem starts in step 1.
Instead of restricting ourselves to split this into parts, can't we just reuse the regex we use for testing whether something is a placeholder and just use it globally?

Reply:
I admit that it is indeed not elegant. It is used due to the replacement of `originalName` with `name`. Consider the following case:
pathPattern: `/path/{param_old}/{param}`
path parameter `param1`: `originalName: param_old` and `name: param`
path parameter `param2`: `originalName: param` and `name: param_new`

Then after executing this function, the pathPattern will be `/path/{param_new}/{param_new}` instead of `/path/{param}/{param_new}`.

This is essentially the test case `parses path template for parameters` in `operation.spec.ts`.

In the previous implementation, {placeholder} is extracted by using split('/'). But I guess we can also use regex `/{[^/?#{}]+}/` to match all the brackets as well and then sort them somehow.